src/lib/y2partitioner/dialogs/main.rb

```diff
@@ -40,7 +43,13 @@ def run(system, initial)
         res = nil
         loop do
           res = super()
-          break if res != :redraw
+
+          next if res == :redraw
+          if res == :abort && Yast::Mode.installation
```

Comment:
I test old yast2-storage and it ask for confirmation also outside of installation. So Please ask for popup always.
BTW do we also have method that compare initial and resulting device graph? because old storage ask only if there are some modifications.